src/app/inbound/mod.rs

```diff
@@ -54,6 +62,11 @@ where
         .push(transport_metrics.connect("inbound"))
         .push(rewrite_loopback_addr::layer());
 
+    let trace_context_layer = trace_context::layer(
+        span_sink
+            .as_ref()
```

Comment:
TIOLI: instead of 
```rust
span_sink
    .as_ref()
    .map(|span_sink| ... span_sink.clone() ... )
```
consider
```rust
span_sink.cloned().map(|span_sink| ... span_sink ...)
```

Reply:
I think this has to be `span_sink.as_ref().cloned().map....` since `span_sink` is not an `Option<&'_ T>`

Reply:
or maybe just `span_sink.clone().map...` ?

Reply:
ah, yeah, you're right --- the last one should work, i think